File: 2436. Minimum Split Into Subarrays With GCD Greater Than One/Solution2436.py
class Solution2436:
    def minimumSplits(self, nums: List[int]) -> int:
        res, gcd = 1, nums[0]
        
        def getGcd(a:int, b:int)->int:
            return a if b == 0 else getGcd(b, a%b)
        
        for num in nums:
            gcd = getGcd(gcd, num)
            if gcd == 1:
                gcd = num
                res += 1
        return res

File: 2436. Minimum Split Into Subarrays With GCD Greater Than One/Solution2436.java
class Solution2436 {
    public int minimumSplits(int[] nums) {
        if (nums == null || nums.length == 0) return 0;
        int gcd = nums[0];
        int res = 1;
        for (int i = 1; i < nums.length; i++) {
            if (getGCD(gcd, nums[i]) == 1) {
                res++;
                gcd = nums[i];
            } else {
                gcd = getGCD(gcd, nums[i]);
            }
        }
        return res;
    }
    public int getGCD(int a, int b) {
        return b == 0 ? a : getGCD(b, a%b);
    }
}